drivers/flash: stm32h7: Fixed range validation

SpooniSpoon opened the pull request:
drivers/flash: stm32h7: Fixed range validation

The range validation routine is not able to detect unaligned memory blocks as it should.  

Signed-off-by: Raphael Löffel <loeffel@rte-ag.ch>

erwango reviewed (changes requested):
Thanks for this change.

Change looks good, but please provide a valid commit message:
https://docs.zephyrproject.org/latest/contribute/index.html#commit-guidelines

erwango commented:
@SpooniSpoon Seems like the PR description is in correct format, but you also need to amend and update the actual commit.

SpooniSpoon commented:
> @SpooniSpoon Seems like the PR description is in correct format, but you also need to amend and update the actual commit.

@erwango OK I tryied to amend the last commit. Unfortunately a merge got committed too afterwards. Is this ok?

erwango commented:
> > @SpooniSpoon Seems like the PR description is in correct format, but you also need to amend and update the actual commit.
> 
> @erwango OK I tryied to amend the last commit. Unfortunately a merge got committed too afterwards. Is this ok?

Unfortunately not. We merge PR branched "as is" on top of zephyr tree, so branch is expected to be clean.

SpooniSpoon commented:
@erwango Hope now its fine. I am familiar with git an therefore have some troubles to get things done the right way.

erwango commented:
> @erwango Hope now its fine. I am familiar with git an therefore have some troubles to get things done the right way.

Well, there are now 5 commits, so situation is worse :-).
It's difficult to help from that point of view.
My suggestion would be to create a new branch from main branch and do your change there,
using the right commit format directly.
Once this is done rename this new branch as 'dev_flash_stm32h7' and force push this branch (this way you don't need to create a new PR).

SpooniSpoon commented:
Hope this works now

erwango commented:
@SpooniSpoon Please have look to issues reported by compiance bot:
https://github.com/zephyrproject-rtos/zephyr/pull/34688/checks?check_run_id=2499274108

SpooniSpoon commented on drivers/flash/flash_stm32h7x.c:
```diff
@@ -100,7 +100,7 @@ bool flash_stm32_valid_range(const struct device *dev, off_t offset,
 
 	if (write) {
-		if ((offset % FLASH_NB_32BITWORD_IN_FLASHWORD * 4) != 0) {
+        if ((offset % (FLASH_NB_32BITWORD_IN_FLASHWORD * 4)) != 0) {
```
It seems that the original file use tabs instead of spaces as defines in the coding style guide.

erwango replied:
I'm not sure to find correct coding guidelines statement, but you should use tabs when possible and fine tune with spaces.
So generally we have tabs to begin a line.
You can check other files to cross check.

SpooniSpoon committed:
drivers/flash: stm32h7: Fixed range validation

The range validation routine is not able to detect unaligned memory
blocks as it should.

Signed-off-by: Raphael Löffel <loeffel@rte-ag.ch>
